### PR TITLE
Add support for Spot Interruption Queue within Karpenter

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

Closes https://github.com/weaveworks/eksctl/issues/6373

Added a new config file field which allows the user to enable Spot Interruption Queue support for Karpenter deployments. This behaviour shouldn't be provided as default since it enforces all AWS Accounts on which Karpenter is deployed to have access to SQS service

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: userdocs/src/usage/eksctl-karpenter.md [hunk 1]
```diff
@@ -2,12 +2,7 @@
 
 `eksctl` provides adding [Karpenter](https://karpenter.sh/) to a newly created cluster. It will create all the necessary
 prerequisites outlined in Karpenter's [Getting Started](https://karpenter.sh/docs/getting-started/getting-started-with-eksctl/) section including installing
```
Comment (Himangini) [inline userdocs/src/usage/eksctl-karpenter.md:4]:
```suggestion
prerequisites outlined in Karpenter's [Getting Started](https://karpenter.sh/docs/getting-started/) section including installing
```
Looks like they have removed the Getting started with eksctl page :/

File: userdocs/src/usage/eksctl-karpenter.md [hunk 2]
```diff
@@ -43,6 +38,7 @@ karpenter:
   version: 'v0.18.0'
```
Comment (Himangini) [inline userdocs/src/usage/eksctl-karpenter.md:38]:
```suggestion
  version: 'v0.20.0'
```

Comment (Himangini) [reply userdocs/src/usage/eksctl-karpenter.md:38]: can we also update the version on line 24

**Review by Himangini — COMMENTED** [review]
Can we also update `examples/29-cluster-with-karpenter.yaml` ?

File: examples/29-cluster-with-karpenter.yaml [hunk 3]
```diff
@@ -12,7 +12,7 @@ iam:
   withOIDC: true
 
 karpenter:
-  version: 'v0.18.0' # Exact version must be provided
+  version: 'v0.20.0' # Exact version must be provided
   createServiceAccount: true # default is false
```
Comment (Himangini) [inline examples/29-cluster-with-karpenter.yaml:16]:
```suggestion
  version: 'v0.20.0' # Exact version must be provided
  createServiceAccount: true # default is false
  withSpotInterruptionQueue: true # adds all required policies and rules for supporting Spot Interruption Queue, default is false
```

Comment (Himangini) [reply examples/29-cluster-with-karpenter.yaml:16]: The version here `integration/tests/karpenter/testdata/cluster-config.yaml` needs to be bumped to `0.20.0`

Comment (TiberiuGC) [reply examples/29-cluster-with-karpenter.yaml:16]: https://github.com/weaveworks/eksctl/pull/6460/commits/8e4b1b7fa25cd34085c650de72f101eb6096f29a

**Review by Himangini — APPROVED** [review]
👍🏻

**Pull request merged**